graphql/queryResolvers/exerciseCrud.ts

```diff
@@ -0,0 +1,78 @@
+import prisma from '../../prisma'
+import {
+  MutationAddExerciseArgs,
+  MutationUpdateExerciseArgs,
+  MutationDeleteExerciseArgs,
+  SuccessResponse
+} from '..'
+import { Context } from '../../@types/helpers'
+import type { Exercise } from '@prisma/client'
+import { isAdmin } from '../../helpers/isAdmin'
+
+export const exercises = async () => {
+  return await prisma.exercise.findMany({
+    include: {
+      author: true,
+      module: true
+    }
+  })
+}
+
+export const addExercise = async (
+  _parent: void,
+  args: MutationAddExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No user')
+  const { testable, testStr, description, answer, moduleId } = args
+  if (testable && !testStr) throw new Error('Testable must have test string')
+  return prisma.exercise.create({
+    data: { authorId, testable, testStr, description, answer, moduleId }
+  })
+}
+
+export const updateExercise = async (
+  _parent: void,
+  args: MutationUpdateExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const authorId = req.user?.id
+  const { id, testable, testStr, description, answer, moduleId } = args
+  if (testable && !testStr) throw new Error('Testable must have test string')
```

Comment:
on second thought, we don't need the `testable` field because we could just check `testStr` is present right (I think this would require a db migration), but db migration is better to be done earlier than later.